cli/.js/src/main/scala/aqua/ArgOpts.scala

```diff
@@ -73,21 +73,44 @@ object ArgOpts {
     (dataFileOrStringOpt[F], funcOpt).mapN { case (dataF, func) =>
       dataF.map { dataV =>
         dataV.andThen { data =>
-          checkDataGetServices(func.args, data).map { getters =>
-            FuncWithData(func, getters)
+          checkDataGetServices(func, data).map { case (funcWithTypedArgs, getters) =>
+            FuncWithData(funcWithTypedArgs, getters)
           }
         }
       }
     }
   }
 
+  // TODO: it is hack, will be deleted after we will have context with types on this stage
+  def jsTypeToAqua(name: String, arg: js.Dynamic): ValidatedNec[String, Type] = {
+    arg match {
+      case a if js.typeOf(a) == "string" => validNec(ScalarType.string)
+      case a if js.typeOf(a) == "number" => validNec(ScalarType.u64)
+      case a if js.typeOf(a) == "boolean" => validNec(ScalarType.bool)
+      case a if js.Array.isArray(a) =>
+        // if all types are similar it will be array array with this type
+        // otherwise array with bottom type
+        val elementsTypesV: ValidatedNec[String, List[Type]] =
+          a.asInstanceOf[js.Array[js.Dynamic]].map(ar => jsTypeToAqua(name, ar)).toList.sequence
+
+        elementsTypesV.andThen { elementsTypes =>
+          if (elementsTypes.isEmpty) validNec(ArrayType(BottomType))
+          else if (elementsTypes.forall(_ == elementsTypes.head))
```

Comment:
What will happen if the array is empty?

Reply:
everything will be ok, cause we already know that it is an array